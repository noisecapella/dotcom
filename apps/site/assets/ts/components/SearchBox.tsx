import React, { ReactElement, useState, useEffect } from "react";
import { searchIcon } from "../helpers/icon";

interface Props {
  id: string;
  labelText: string;
  onChange: (q: string) => void;
  placeholder?: string;
  className?: string;
}

const SearchBox = ({
  id,
  labelText,
  onChange,
  placeholder,
  className
}: Props): ReactElement<HTMLElement> | null => {
  const [query, setQuery] = useState("");

  useEffect(() => {
    onChange(query);
  }, [query, onChange]);

  return (
    <div className={`${className} c-form__input-container`}>
      <label htmlFor={id} className="sr-only">
        {labelText}
      </label>
      <input
        className="c-form__input"
        id={id}
        type="text"
        onChange={event => {
          setQuery(event.target.value);
        }}
        value={query}
        placeholder={placeholder || labelText}
      />
      {query !== "" && (
        <button
          className="c-form__reset-btn btn-link"
          aria-label="reset"
          type="button"
          onClick={() => {
            setQuery("");
          }}
        >
          <i
            aria-hidden="true"
            className="c-form__reset-icon fa fa-times-circle"
          />
        </button>
      )}
      <button
        className="c-form__submit-btn"
        aria-label="search"
        type="button"
        onClick={() => {
          setQuery(query);
        }}
      >
        {searchIcon("c-svg__icon-search-reverse-default")}
      </button>
    </div>
  );
};

export default SearchBox;
