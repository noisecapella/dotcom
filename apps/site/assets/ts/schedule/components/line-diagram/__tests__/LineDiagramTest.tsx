import React from "react";
import * as swr from "swr";
import { mount, ReactWrapper } from "enzyme";
import { cloneDeep, merge } from "lodash";
import LineDiagramAndStopListPage from "../LineDiagram";
import { EnhancedRoute, RouteType } from "../../../../__v3api";
import {
  LineDiagramStop,
  RoutePatternsByDirection,
  SimpleStop
} from "../../__schedule";
import * as routePatternsByDirection from "../../__tests__/test-data/routePatternsByDirectionData.json";
import simpleLineDiagram from "./lineDiagramData/simple.json"; // not a full line diagram
import outwardLineDiagram from "./lineDiagramData/outward.json"; // not a full line diagram
import simpleLiveData from "./lineDiagramData/live-data.json";
import SearchBox from "../../../../components/SearchBox";

const lineDiagram = (simpleLineDiagram as unknown) as LineDiagramStop[];
let lineDiagramBranchingOut = (outwardLineDiagram as unknown) as LineDiagramStop[];

// Mock useSWR to return fixture data
jest.mock("swr", () => {
  return jest.fn(() => ({ data: simpleLiveData }));
});

const route = {
  type: 3 as RouteType,
  name: "route 1",
  long_name: "route 1 long name",
  color: "F00B42",
  id: "route-1",
  direction_names: {
    0: "Outbound",
    1: "Inbound"
  },
  direction_destinations: {
    0: "Begin",
    1: "End"
  },
  description: "key_bus_route",
  "custom_route?": false,
  header: "",
  alerts: []
};

lineDiagram.forEach(({ route_stop }) => {
  route_stop.route = cloneDeep(route);
});

lineDiagramBranchingOut.forEach(({ route_stop }) => {
  route_stop.route = cloneDeep(route);
});

let lineDiagramBranchingIn = cloneDeep(lineDiagramBranchingOut).reverse();
const CRroute = merge(cloneDeep(route), { type: 2 as RouteType });
lineDiagramBranchingIn.forEach(({ route_stop }) => {
  route_stop.route = CRroute;
  if (route_stop["is_terminus?"]) {
    route_stop["is_beginning?"] = !route_stop["is_beginning?"];
  }
});

const stops = lineDiagram.map(({ route_stop }) => ({
  name: route_stop.name,
  id: route_stop.id,
  is_closed: false,
  zone: route_stop.zone || null
})) as SimpleStop[];

const directionId = 1;

describe("LineDiagram", () => {
  let wrapper: ReactWrapper;
  beforeEach(() => {
    wrapper = mount(
      <LineDiagramAndStopListPage
        lineDiagram={lineDiagram}
        route={route as EnhancedRoute}
        directionId={directionId}
      />
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("renders and matches snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("can filter stops by name", () => {
    const filter = wrapper.find(".m-schedule-diagram__filter").at(0);
    expect(filter.exists()).toBeTruthy();
    expect(filter.type()).toEqual(SearchBox);
  });

  it.skip("includes buttons to open the Schedule Finder modal", () => {
    expect(wrapper.exists(".schedule-finder--modal")).toBeFalsy();

    wrapper
      .find(".m-schedule-diagram__footer > button")
      .first()
      .simulate("click");

    expect(wrapper.exists(".schedule-finder--modal")).toBeTruthy();

    expect(wrapper.exists("#modal-close")).toBeTruthy();

    wrapper.find("#modal-close").simulate("click");
    expect(wrapper.exists(".schedule-finder--modal")).toBeFalsy();
    expect(wrapper.exists("#modal-close")).toBeFalsy();
  });

  describe("opens the ScheduleFinderModal", () => {
    it.skip("detects a change in direction (and hence in origin)", () => {
      // open modal:
      wrapper
        .find(".m-schedule-diagram__footer > button")
        .first()
        .simulate("click");

      // change direction
      wrapper
        .find("select")
        .at(0)
        .simulate("change", { target: { value: "0" } });
    });

    it.skip("detects a change in origin", () => {
      // open modal:
      wrapper
        .find(".m-schedule-diagram__footer > button")
        .first()
        .simulate("click");

      // change origin
      wrapper
        .find("select")
        .at(1)
        .simulate("change", { target: { value: "line-stop2" } });
    });

    it.skip("detects an origin selection", () => {
      // open modal:
      wrapper
        .find(".m-schedule-diagram__footer > button")
        .first()
        .simulate("click");

      // Click on the SelectContainer for the origin select
      wrapper
        .find("SelectContainer")
        .last()
        // @ts-ignore -- types for `invoke` are too restrictive?
        .invoke("handleClick")();
    });
  });
});

it.each`
  type | name
  ${0} | ${"Stations"}
  ${1} | ${"Stations"}
  ${2} | ${"Stations"}
  ${3} | ${"Stops"}
  ${4} | ${"Stops"}
`(
  "LineDiagram names stops or stations for route type $type",
  ({ type, name }) => {
    const wrapper = mount(
      <LineDiagramAndStopListPage
        lineDiagram={lineDiagram}
        route={
          {
            ...route,
            type: type as RouteType
          } as EnhancedRoute
        }
        directionId={directionId}
      />
    );
    expect(wrapper.find(".m-schedule-diagram__heading").text()).toContain(name);
  }
);

it.each`
  type | willPoll
  ${0} | ${true}
  ${1} | ${true}
  ${2} | ${true}
  ${3} | ${true}
  ${4} | ${false}
`(
  "LineDiagram requests live data for most route types: $type",
  ({ type, willPoll }) => {
    // don't mock the return value here,
    // we just want to check if it's called
    const useSWRSpy = jest.spyOn(swr, "default");
    const wrapper = mount(
      <LineDiagramAndStopListPage
        lineDiagram={lineDiagram}
        route={
          {
            ...route,
            type: type as RouteType
          } as EnhancedRoute
        }
        directionId={directionId}
      />
    );
    expect(useSWRSpy).toHaveBeenCalled();

    if (willPoll) {
      expect(useSWRSpy).toHaveBeenCalledWith(
        "/schedules/line_api/realtime?id=route-1&direction_id=1",
        expect.any(Function),
        expect.objectContaining({ refreshInterval: expect.any(Number) })
      );
    } else {
      // will not ping the realtime endpoint
      expect(useSWRSpy).toHaveBeenCalledWith(
        "",
        expect.any(Function),
        expect.objectContaining({
          refreshInterval: expect.any(Number)
        })
      );
    }
  }
);
