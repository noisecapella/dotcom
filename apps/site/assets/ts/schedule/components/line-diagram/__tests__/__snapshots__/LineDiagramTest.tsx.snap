// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineDiagram renders and matches snapshot 1`] = `
"<LineDiagramAndStopListPage lineDiagram={{...}} route={{...}} directionId={1} routePatternsByDirection={{...}} services={{...}} stops={{...}} today=\\"2019-12-05\\" scheduleNote={{...}}>
  <h3 className=\\"m-schedule-diagram__heading\\">
    Stops
  </h3>
  <SearchBox id=\\"stop-search\\" labelText=\\"Search for a stop\\" onChange={[Function: bound dispatchAction]} className=\\"m-schedule-diagram__filter\\">
    <div className=\\"m-schedule-diagram__filter c-form__input-container\\">
      <label htmlFor=\\"stop-search\\" className=\\"sr-only\\">
        Search for a stop
      </label>
      <input className=\\"c-form__input\\" id=\\"stop-search\\" type=\\"text\\" onChange={[Function: onChange]} value=\\"\\" placeholder=\\"Search for a stop\\" />
      <button className=\\"c-form__submit-btn\\" aria-label=\\"search\\" type=\\"button\\" onClick={[Function: onClick]}>
        <span className=\\"notranslate c-svg__icon-search-reverse-default\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
      </button>
    </div>
  </SearchBox>
  <Provider store={{...}}>
    <LineDiagramWithStops stops={{...}} handleStopClick={[Function]} liveData={{...}}>
      <div className=\\"m-schedule-diagram u-no-crowding-data\\">
        <Diagram lineDiagram={{...}} liveData={{...}}>
          <LiveVehicleIconSet stop={{...}} liveData={{...}} />
          <LiveVehicleIconSet stop={{...}} liveData={{...}}>
            <VehicleIcons stop={{...}} vehicles={{...}}>
              <div className=\\"m-schedule-diagram__vehicle\\" style={{...}}>
                <TooltipWrapper tooltipText=\\"<div class=\\"m-schedule-diagram__vehicle-tooltip\\">tooltip for vehicle 1 at stop 1</div>\\" tooltipOptions={{...}}>
                  <span href={[undefined]} data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"right\\" data-animation={false} data-html={true} data-selector=\\"true\\" data-original-title=\\"<div class=\\"m-schedule-diagram__vehicle-tooltip\\">tooltip for vehicle 1 at stop 1</div>\\" title=\\"<div class=\\"m-schedule-diagram__vehicle-tooltip\\">tooltip for vehicle 1 at stop 1</div>\\">
                    <span className=\\"notranslate m-schedule-diagram__vehicle--icon\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                  </span>
                </TooltipWrapper>
              </div>
              <div className=\\"m-schedule-diagram__vehicle\\" style={{...}}>
                <TooltipWrapper tooltipText=\\"<div class=\\"m-schedule-diagram__vehicle-tooltip\\">tooltip for vehicle 2 at stop 1</div>\\" tooltipOptions={{...}}>
                  <span href={[undefined]} data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"right\\" data-animation={false} data-html={true} data-selector=\\"true\\" data-original-title=\\"<div class=\\"m-schedule-diagram__vehicle-tooltip\\">tooltip for vehicle 2 at stop 1</div>\\" title=\\"<div class=\\"m-schedule-diagram__vehicle-tooltip\\">tooltip for vehicle 2 at stop 1</div>\\">
                    <span className=\\"notranslate m-schedule-diagram__vehicle--icon\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                  </span>
                </TooltipWrapper>
              </div>
            </VehicleIcons>
          </LiveVehicleIconSet>
          <LiveVehicleIconSet stop={{...}} liveData={{...}}>
            <VehicleIcons stop={{...}} vehicles={{...}}>
              <div className=\\"m-schedule-diagram__vehicle\\" style={{...}}>
                <TooltipWrapper tooltipText=\\"<div class=\\"m-schedule-diagram__vehicle-tooltip\\">tooltip for vehicle 3 at stop 2</div>\\" tooltipOptions={{...}}>
                  <span href={[undefined]} data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"right\\" data-animation={false} data-html={true} data-selector=\\"true\\" data-original-title=\\"<div class=\\"m-schedule-diagram__vehicle-tooltip\\">tooltip for vehicle 3 at stop 2</div>\\" title=\\"<div class=\\"m-schedule-diagram__vehicle-tooltip\\">tooltip for vehicle 3 at stop 2</div>\\">
                    <span className=\\"notranslate m-schedule-diagram__vehicle--icon\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                  </span>
                </TooltipWrapper>
              </div>
            </VehicleIcons>
          </LiveVehicleIconSet>
          <LiveVehicleIconSet stop={{...}} liveData={{...}} />
          <svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" aria-labelledby=\\"diagram-title diagram-desc\\" className=\\"line-diagram-svg bus\\" width=\\"40px\\" height=\\"100%\\" style={{...}}>
            <title id=\\"diagram-title\\">
              Line diagram for
               
              route 1
            </title>
            <desc id=\\"diagram-desc\\">
              4 stops going Outbound to Begin
            </desc>
            <defs>
              <pattern id=\\"diagonalHatch\\" width=\\"10\\" height=\\"10\\" patternTransform=\\"rotate(45 0 0)\\" patternUnits=\\"userSpaceOnUse\\">
                <line className=\\"line-diagram-svg__line\\" y2={8} strokeWidth=\\"16px\\" />
              </pattern>
            </defs>
            <Line from={{...}} to={{...}}>
              <line className=\\"line-diagram-svg__line\\" strokeWidth=\\"10px\\" x1=\\"11px\\" x2=\\"11px\\" y1=\\"0px\\" y2=\\"0px\\" stroke=\\"url(#diagonalHatch)\\" />
            </Line>
            <Line from={{...}} to={{...}}>
              <line className=\\"line-diagram-svg__line\\" strokeWidth=\\"10px\\" x1=\\"11px\\" x2=\\"11px\\" y1=\\"0px\\" y2=\\"0px\\" />
            </Line>
            <Line from={{...}} to={{...}}>
              <line className=\\"line-diagram-svg__line\\" strokeWidth=\\"10px\\" x1=\\"11px\\" x2=\\"11px\\" y1=\\"0px\\" y2=\\"0px\\" />
            </Line>
            <Stop stop={{...}}>
              <circle className=\\"line-diagram-svg__stop\\" r=\\"4px\\" cx=\\"11px\\" cy=\\"0px\\" />
            </Stop>
            <Stop stop={{...}}>
              <circle className=\\"line-diagram-svg__stop\\" r=\\"4px\\" cx=\\"11px\\" cy=\\"0px\\" />
            </Stop>
            <Stop stop={{...}}>
              <circle className=\\"line-diagram-svg__stop\\" r=\\"4px\\" cx=\\"11px\\" cy=\\"0px\\" />
            </Stop>
            <Stop stop={{...}}>
              <circle className=\\"line-diagram-svg__stop\\" r=\\"4px\\" cx=\\"11px\\" cy=\\"0px\\" />
            </Stop>
          </svg>
        </Diagram>
        <ol>
          <StopCard stop={{...}} onClick={[Function]} liveData={[undefined]}>
            <li className=\\"m-schedule-diagram__stop\\" style={{...}}>
              <section className=\\"m-schedule-diagram__content\\">
                <header className=\\"m-schedule-diagram__stop-heading\\">
                  <h4 className=\\"m-schedule-diagram__stop-link\\">
                    <a href=\\"/stops/line-origin\\">
                      <MatchHighlight text=\\"Origin\\" matchQuery={[undefined]}>
                        <span>
                          Origin
                        </span>
                      </MatchHighlight>
                    </a>
                  </h4>
                  <div className=\\"m-schedule-diagram__features\\">
                    <TooltipWrapper tooltipText=\\"Parking\\" tooltipOptions={{...}}>
                      <span href={[undefined]} data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"bottom\\" data-animation={true} data-html={false} data-selector=\\"true\\" data-original-title=\\"Parking\\" title=\\"Parking\\">
                        <span className=\\"notranslate c-svg__icon-parking-default m-schedule-diagram__feature-icon\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                      </span>
                    </TooltipWrapper>
                  </div>
                </header>
                <div className=\\"m-schedule-diagram__stop-details\\">
                  <div className=\\"m-schedule-diagram__connections\\" />
                  <div className=\\"m-schedule-diagram__alert\\">
                    Detour
                  </div>
                </div>
                <footer className=\\"m-schedule-diagram__footer\\">
                  <button className=\\"btn btn-link\\" type=\\"button\\" onClick={[Function: onClick]}>
                    View schedule
                  </button>
                </footer>
              </section>
            </li>
          </StopCard>
          <StopCard stop={{...}} onClick={[Function]} liveData={{...}}>
            <li className=\\"m-schedule-diagram__stop\\" style={{...}}>
              <section className=\\"m-schedule-diagram__content\\">
                <header className=\\"m-schedule-diagram__stop-heading\\">
                  <h4 className=\\"m-schedule-diagram__stop-link\\">
                    <a href=\\"/stops/line-stop1\\">
                      <span className=\\"notranslate c-svg__icon-alerts-triangle\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                      <span className=\\"sr-only\\">
                        Service alert or delay
                      </span>
                      Â 
                      <MatchHighlight text=\\"Stop 1\\" matchQuery={[undefined]}>
                        <span>
                          Stop 1
                        </span>
                      </MatchHighlight>
                    </a>
                  </h4>
                  <div className=\\"m-schedule-diagram__features\\" />
                </header>
                <div className=\\"m-schedule-diagram__stop-details\\">
                  <div className=\\"m-schedule-diagram__connections\\">
                    <TooltipWrapper href=\\"/schedules/Orange/line\\" tooltipText=\\"Orange Line\\" tooltipOptions={{...}}>
                      <a href=\\"/schedules/Orange/line\\" data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"bottom\\" data-animation={false} data-html={false} data-selector=\\"true\\" data-original-title=\\"Orange Line\\" title=\\"Orange Line\\">
                        <span className=\\"m-schedule-diagram__connection\\">
                          <span className=\\"notranslate c-svg__icon\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                        </span>
                      </a>
                    </TooltipWrapper>
                    <TooltipWrapper href=\\"/schedules/Green-C/line\\" tooltipText=\\"Green Line C\\" tooltipOptions={{...}}>
                      <a href=\\"/schedules/Green-C/line\\" data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"bottom\\" data-animation={false} data-html={false} data-selector=\\"true\\" data-original-title=\\"Green Line C\\" title=\\"Green Line C\\">
                        <span className=\\"m-schedule-diagram__connection\\">
                          <span className=\\"notranslate c-svg__icon\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                        </span>
                      </a>
                    </TooltipWrapper>
                  </div>
                  <StopPredictions headsigns={{...}} isCommuterRail={false}>
                    <div className=\\"m-schedule-diagram__predictions\\">
                      <div className=\\"m-schedule-diagram__prediction\\">
                        <div>
                          Dest1
                        </div>
                        <div className=\\"m-schedule-diagram__prediction-time\\">
                          2   min
                        </div>
                        <LiveCrowdingIcon crowding={{...}}>
                          <div className=\\"m-schedule-diagram__prediction-crowding\\">
                            <span className=\\"notranslate c-svg__icon c-icon__crowding c-icon__crowding--crowding_unavailable\\" aria-hidden=\\"true\\" dangerouslySetInnerHTML={{...}} />
                          </div>
                        </LiveCrowdingIcon>
                      </div>
                      <div className=\\"m-schedule-diagram__prediction\\">
                        <div>
                          Dest2
                        </div>
                        <div className=\\"m-schedule-diagram__prediction-time\\">
                          5   min
                        </div>
                        <LiveCrowdingIcon crowding={{...}}>
                          <div className=\\"m-schedule-diagram__prediction-crowding\\">
                            <span className=\\"notranslate c-svg__icon c-icon__crowding c-icon__crowding--crowding_unavailable\\" aria-hidden=\\"true\\" dangerouslySetInnerHTML={{...}} />
                          </div>
                        </LiveCrowdingIcon>
                      </div>
                    </div>
                  </StopPredictions>
                </div>
                <footer className=\\"m-schedule-diagram__footer\\">
                  <button className=\\"btn btn-link\\" type=\\"button\\" onClick={[Function: onClick]}>
                    View schedule
                  </button>
                </footer>
              </section>
            </li>
          </StopCard>
          <StopCard stop={{...}} onClick={[Function]} liveData={{...}}>
            <li className=\\"m-schedule-diagram__stop\\" style={{...}}>
              <section className=\\"m-schedule-diagram__content\\">
                <header className=\\"m-schedule-diagram__stop-heading\\">
                  <h4 className=\\"m-schedule-diagram__stop-link\\">
                    <a href=\\"/stops/line-stop2\\">
                      <span className=\\"notranslate c-svg__icon-alerts-triangle\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                      <span className=\\"sr-only\\">
                        Service alert or delay
                      </span>
                      Â 
                      <MatchHighlight text=\\"Stop 2\\" matchQuery={[undefined]}>
                        <span>
                          Stop 2
                        </span>
                      </MatchHighlight>
                    </a>
                  </h4>
                  <div className=\\"m-schedule-diagram__features\\">
                    <TooltipWrapper tooltipText=\\"Accessible\\" tooltipOptions={{...}}>
                      <span href={[undefined]} data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"bottom\\" data-animation={true} data-html={false} data-selector=\\"true\\" data-original-title=\\"Accessible\\" title=\\"Accessible\\">
                        <span className=\\"notranslate c-svg__icon-acessible-default m-schedule-diagram__feature-icon\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                      </span>
                    </TooltipWrapper>
                  </div>
                </header>
                <div className=\\"m-schedule-diagram__stop-details\\">
                  <div className=\\"m-schedule-diagram__connections\\">
                    <TooltipWrapper href=\\"/schedules/62/line\\" tooltipText=\\"Route 62\\" tooltipOptions={{...}}>
                      <a href=\\"/schedules/62/line\\" data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"bottom\\" data-animation={false} data-html={false} data-selector=\\"true\\" data-original-title=\\"Route 62\\" title=\\"Route 62\\">
                        <span className=\\"c-icon__bus-pill--small m-schedule-diagram__connection u-bg--bus\\">
                          62
                        </span>
                      </a>
                    </TooltipWrapper>
                    <TooltipWrapper href=\\"/schedules/67/line\\" tooltipText=\\"Route 67\\" tooltipOptions={{...}}>
                      <a href=\\"/schedules/67/line\\" data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"bottom\\" data-animation={false} data-html={false} data-selector=\\"true\\" data-original-title=\\"Route 67\\" title=\\"Route 67\\">
                        <span className=\\"c-icon__bus-pill--small m-schedule-diagram__connection u-bg--bus\\">
                          67
                        </span>
                      </a>
                    </TooltipWrapper>
                  </div>
                  <StopPredictions headsigns={{...}} isCommuterRail={false}>
                    <div className=\\"m-schedule-diagram__predictions\\">
                      <div className=\\"m-schedule-diagram__prediction\\">
                        <div>
                          Dest2
                        </div>
                        <div className=\\"m-schedule-diagram__prediction-time\\">
                          7   min
                        </div>
                        <LiveCrowdingIcon crowding={{...}}>
                          <div className=\\"m-schedule-diagram__prediction-crowding\\">
                            <span className=\\"notranslate c-svg__icon c-icon__crowding c-icon__crowding--crowding_unavailable\\" aria-hidden=\\"true\\" dangerouslySetInnerHTML={{...}} />
                          </div>
                        </LiveCrowdingIcon>
                      </div>
                    </div>
                  </StopPredictions>
                </div>
                <footer className=\\"m-schedule-diagram__footer\\">
                  <button className=\\"btn btn-link\\" type=\\"button\\" onClick={[Function: onClick]}>
                    View schedule
                  </button>
                </footer>
              </section>
            </li>
          </StopCard>
          <StopCard stop={{...}} onClick={[Function]} liveData={[undefined]}>
            <li className=\\"m-schedule-diagram__stop\\" style={{...}}>
              <section className=\\"m-schedule-diagram__content\\">
                <header className=\\"m-schedule-diagram__stop-heading\\">
                  <h4 className=\\"m-schedule-diagram__stop-link\\">
                    <a href=\\"/stops/line-destination\\">
                      <MatchHighlight text=\\"Destination\\" matchQuery={[undefined]}>
                        <span>
                          Destination
                        </span>
                      </MatchHighlight>
                    </a>
                  </h4>
                  <div className=\\"m-schedule-diagram__features\\">
                    <TooltipWrapper tooltipText=\\"Parking\\" tooltipOptions={{...}}>
                      <span href={[undefined]} data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"bottom\\" data-animation={true} data-html={false} data-selector=\\"true\\" data-original-title=\\"Parking\\" title=\\"Parking\\">
                        <span className=\\"notranslate c-svg__icon-parking-default m-schedule-diagram__feature-icon\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                      </span>
                    </TooltipWrapper>
                    <TooltipWrapper tooltipText=\\"Accessible\\" tooltipOptions={{...}}>
                      <span href={[undefined]} data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"bottom\\" data-animation={true} data-html={false} data-selector=\\"true\\" data-original-title=\\"Accessible\\" title=\\"Accessible\\">
                        <span className=\\"notranslate c-svg__icon-acessible-default m-schedule-diagram__feature-icon\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                      </span>
                    </TooltipWrapper>
                  </div>
                </header>
                <div className=\\"m-schedule-diagram__stop-details\\">
                  <div className=\\"m-schedule-diagram__connections\\">
                    <TooltipWrapper href=\\"/schedules/BOAT-A/line\\" tooltipText=\\"Atlantis\\" tooltipOptions={{...}}>
                      <a href=\\"/schedules/BOAT-A/line\\" data-toggle=\\"tooltip\\" data-trigger=\\"hover focus\\" data-placement=\\"bottom\\" data-animation={false} data-html={false} data-selector=\\"true\\" data-original-title=\\"Atlantis\\" title=\\"Atlantis\\">
                        <span className=\\"m-schedule-diagram__connection\\">
                          <span className=\\"notranslate c-svg__icon-bus-small\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                        </span>
                      </a>
                    </TooltipWrapper>
                  </div>
                </div>
                <footer className=\\"m-schedule-diagram__footer\\">
                  <button className=\\"btn btn-link\\" type=\\"button\\" onClick={[Function: onClick]}>
                    View schedule
                  </button>
                </footer>
              </section>
            </li>
          </StopCard>
        </ol>
      </div>
    </LineDiagramWithStops>
  </Provider>
</LineDiagramAndStopListPage>"
`;
